### PR TITLE
docs: quick-start add docker alternatives

### LINKED ISSUE #1048: Getting started guide to include an alternative to docker-desktop
We have to give an alternative to docker-desktop. Perhaps, provide podman documentation (#206)

### PR DESCRIPTION
**Pull request by tczhao**
Fixes #1048

### CODE REVIEW

File: docs/quick-start.md [hunk 1]
```diff
@@ -7,7 +7,15 @@ In this page, we will guide you through the steps to:
 
 ## Before you begin: prerequisites
 
-To try Numaflow, you will first need to create a local cluster. This can be done simply using [Docker Desktop](https://www.docker.com/blog/how-kubernetes-works-under-the-hood-with-docker-desktop). Follow the steps on [Docker's documentation](https://docs.docker.com/desktop/kubernetes/) to learn how to create a local Kubernetes cluster.
+To try Numaflow, you will first need to setup using one of the following options to run container images:
+- [Docker Desktop](https://www.docker.com/blog/how-kubernetes-works-under-the-hood-with-docker-desktop)
```
Comment (whynowy) [inline docs/quick-start.md:11]: Using the link below for downloading might be better.

https://docs.docker.com/get-docker/

File: docs/quick-start.md [hunk 2]
```diff
@@ -7,7 +7,15 @@ In this page, we will guide you through the steps to:
...
+- [podman](https://podman.io/)
+
+Then use one of the following options to create a local Kubernete Cluster:
+- [Docker Desktop](https://docs.docker.com/desktop/kubernetes/)
```
Comment (whynowy) [inline docs/quick-start.md:15]: How about using `Docker Desktop Kubernetes` to differentiate from the `Docker Desktop` above?

File: docs/quick-start.md (hunk 1 above)
Comment (tczhao) [reply docs/quick-start.md]: updated

File: docs/quick-start.md (hunk 2 above)
Comment (tczhao) [reply docs/quick-start.md]: Good point

**Review by whynowy — APPROVED** [review]

**Pull request merged**